com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKLineVisual.cs

```diff
@@ -367,14 +367,14 @@ private void UpdateLineVisual()
                         {
                             lineRenderer.colorGradient = ColorUtilities.GradientLerp(ValidColorGradient, SelectActiveColorGradient, rayInteractor.hasSelection ? 1 : 0);
                         }
+
+                        var compressionAmount = Mathf.Clamp(rayInteractor.maxRaycastDistance * MaxGradientLength / hitDistance, 0.0f, 1.0f);
+                        lineRenderer.colorGradient = ColorUtilities.GradientCompress(lineRenderer.colorGradient, 0.0f, compressionAmount);
```

Comment:
instead of only calling this less frequently (or in addition to), can we make this method not alloc at all?

Reply:
After much investigation, I learned a few things.

1. calls to retrieve lineRender.colorGradient allocs a new instance of Gradient.
2. Calls to retrieve gradient.colorKeys and gradient.alpha keys allocs new arrays.

With these considerations, we can't do much to the fundamental implementation of our utility functions to avoid these GC allocs. I've made other adjustments to the LineVisuals implementation to avoid calling these functions when not necessary, but fundamentally doing modifications of gradients will end up being costly due to the underlying native implemenation.

Reply:
If it's fundamentally unsolvable, we should be doing this in the shader.

Reply:
Though I agree, I think that the performance hit (a few allocs during inbetween time of selection) doesn't quite justify the engineering effort for a rework at this particular juncture. It's not regression from what we've had  previously in MRTK2, so I don't think it should have much customer facing impact.